wpilibj/src/shared/java/edu/wpi/first/wpilibj/command/Scheduler.java

```diff
@@ -274,9 +237,8 @@ void remove(Command command) {
    */
   public void removeAll() {
     // TODO: Confirm that this works with "uninteruptible" commands
-    while (m_firstCommand != null) {
-      remove(m_firstCommand.getData());
-    }
+    // Use a temporary view to avoid concurrent modification
+    new ArrayList<>(m_commands).forEach(this::remove);
```

Comment:
Really hate the temp allocation. Perhaps we should have a private overload of remove that doesn't do the remove from `m_commands`. Teams can have quite a few commands running at any given time, and this would be low-hanging fruit for needless allocation.